### PR TITLE
Unquarantine tests

### LINKED ISSUE #34685: Test failure: StreamPool_MultipleStreamsInSequence_PooledStreamReused
## Failing Test(s)

Microsoft.AspNetCore.Server.Kestrel.Core.Tests.Http3ConnectionTests.StreamPool_MultipleStreamsInSequence_PooledStreamReused

## Error Message

```text
Assert.False() Failure
Expected: False
Actual:   True
```

## Stacktrace

<details>

```text
   at Microsoft.AspNetCore.Server.Kestrel.Core.Tests.Http3ConnectionTests.StreamPool_MultipleStreamsInSequence_PooledStreamReused() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3ConnectionTests.cs:line 247
--- End of stack trace from previous location ---
```
</details>

## Logs

<details>

```text
| [0.000s] TestLifetime Information: Starting test StreamPool_MultipleStreamsInSequence_PooledStreamReused at 2021-07-24T13:12:39
| [0.087s] Microsoft.AspNetCore.Server.Kestrel.Http3 Trace: Connection id "(null)" received HEADERS frame for stream ID 4 with length 65.
| [0.087s] Microsoft.AspNetCore.Server.Kestrel.Http3 Trace: Connection id "(null)" received DATA frame for stream ID 4 with length 13.
| [0.117s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "(null)", Request id "(null)": started reading request body.
| [0.117s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "(null)", Request id "(null)": done reading request body.
| [0.117s] Microsoft.AspNetCore.Server.Kestrel.Http3 Trace: Connection id "(null)" sending HEADERS frame for stream ID 4 with length 38.
| [0.117s] Microsoft.AspNetCore.Server.Kestrel.Http3 Trace: Connection id "(null)" sending DATA frame for stream ID 4 with length 13.
| [0.289s] Microsoft.AspNetCore.Server.Kestrel.Http3 Trace: Connection id "(null)" received HEADERS frame for stream ID 8 with length 65.
| [0.289s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "(null)", Request id "(null)": started reading request body.
| [0.290s] Microsoft.AspNetCore.Server.Kestrel.Http3 Trace: Connection id "(null)" received DATA frame for stream ID 8 with length 13.
| [0.290s] Microsoft.AspNetCore.Server.Kestrel.Http3 Trace: Connection id "(null)" sending HEADERS frame for stream ID 8 with length 38.
| [0.290s] Microsoft.AspNetCore.Server.Kestrel.Http3 Trace: Connection id "(null)" sending DATA frame for stream ID 8 with length 13.
| [0.290s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "(null)", Request id "(null)": done reading request body.
| [0.563s] Microsoft.AspNetCore.Server.Kestrel.Core.Tests.Http3ConnectionTests Error: Test threw an exception.
| Xunit.Sdk.FalseException: Assert.False() Failure
| Expected: False
| Actual:   True
|    at Xunit.Assert.False(Nullable`1 condition, String userMessage) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\BooleanAsserts.cs:line 52
|    at Xunit.Assert.False(Boolean condition) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\BooleanAsserts.cs:line 19
|    at Microsoft.AspNetCore.Server.Kestrel.Core.Tests.Http3ConnectionTests.StreamPool_MultipleStreamsInSequence_PooledStreamReused() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3ConnectionTests.cs:line 247
|    at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 273
| --- End of stack trace from previous location ---
|    at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 54
|    at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\Dev\xunit\xunit\src\xunit.core\Sdk\ExceptionAggregator.cs:line 96
| [0.566s] TestLifetime Information: Finished test StreamPool_MultipleStreamsInSequence_PooledStreamReused in 0.5656032s
```
</details>

## Build

https://dev.azure.com/dnceng/public/_build/results?buildId=1256978&view=ms.vss-test-web.build-test-results-tab&runId=37284574&resultId=106365&paneView=debug

### LINKED ISSUE #34903: DATA_Received_TooSlowlyOnSecondStream_AbortsConnectionAfterNonAdditiveRateTimeout
## Failing Test(s)

- Microsoft.AspNetCore.Server.Kestrel.Core.Tests.Http3TimeoutTests.DATA_Received_TooSlowlyOnSecondStream_AbortsConnectionAfterNonAdditiveRateTimeout

## Error Message

```text
System.InvalidOperationException : No data received.
```

## Stacktrace

<details>

```text
at Microsoft.AspNetCore.Testing.Http3StreamBase.ReceiveFrameAsync(Boolean expectEnd, Boolean allowEnd, Http3FrameWithPayload frame) in /_/src/Servers/Kestrel/shared/test/Http3/Http3InMemory.cs:line 495
   at Microsoft.AspNetCore.Testing.Http3RequestStream.ExpectHeadersAsync(Boolean expectEnd) in /_/src/Servers/Kestrel/shared/test/Http3/Http3InMemory.cs:line 650
   at Microsoft.AspNetCore.Server.Kestrel.Core.Tests.Http3TimeoutTests.DATA_Received_TooSlowlyOnSecondStream_AbortsConnectionAfterNonAdditiveRateTimeout() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3TimeoutTests.cs:line 520
--- End of stack trace from previous location ---
```
</details>

## Logs

<details>

```text
[0.001s] [TestLifetime] [Information] Starting test DATA_Received_TooSlowlyOnSecondStream_AbortsConnectionAfterNonAdditiveRateTimeout at 2021-07-31T03:17:41
[0.007s] [Microsoft.AspNetCore.Testing.Http3InMemory] [Information] Initializing stream 3
[0.007s] [Microsoft.AspNetCore.Testing.Http3InMemory] [Information] Initializing stream 4
[0.018s] [Microsoft.AspNetCore.Server.Kestrel.Http3] [Verbose] Connection id "null" received HEADERS frame for stream ID 4 with length 68.
[0.018s] [Microsoft.AspNetCore.Server.Kestrel.Http3] [Verbose] Connection id "null" received DATA frame for stream ID 4 with length 16384.
[0.019s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "null", Request id "TEST:4": started reading request body.
[0.019s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "null", Request id "TEST:4": done reading request body.
[0.019s] [Microsoft.AspNetCore.Server.Kestrel.Http3] [Verbose] Connection id "null" sending HEADERS frame for stream ID 4 with length 38.
[0.019s] [Microsoft.AspNetCore.Server.Kestrel.Http3] [Verbose] Connection id "null" sending DATA frame for stream ID 4 with length 1.
[0.020s] [Microsoft.AspNetCore.Testing.Http3InMemory] [Information] Disposing stream 4
[0.025s] [Microsoft.AspNetCore.Testing.Http3InMemory] [Information] Initializing stream 8
[0.030s] [Microsoft.AspNetCore.Server.Kestrel] [Information] Connection id "null", Request id "TEST:8": the application completed without reading the entire request body.
[0.031s] [Microsoft.AspNetCore.Testing.Http3InMemory] [Information] Disposing stream 8
[0.048s] [Microsoft.AspNetCore.Server.Kestrel.Core.Tests.Http3TimeoutTests] [Error] Test threw an exception.
System.InvalidOperationException: No data received.
   at Microsoft.AspNetCore.Testing.Http3StreamBase.ReceiveFrameAsync(Boolean expectEnd, Boolean allowEnd, Http3FrameWithPayload frame) in /_/src/Servers/Kestrel/shared/test/Http3/Http3InMemory.cs:line 495
   at Microsoft.AspNetCore.Testing.Http3RequestStream.ExpectHeadersAsync(Boolean expectEnd) in /_/src/Servers/Kestrel/shared/test/Http3/Http3InMemory.cs:line 650
   at Microsoft.AspNetCore.Server.Kestrel.Core.Tests.Http3TimeoutTests.DATA_Received_TooSlowlyOnSecondStream_AbortsConnectionAfterNonAdditiveRateTimeout() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3TimeoutTests.cs:line 520
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 273
--- End of stack trace from previous location ---
   at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 54
   at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\Dev\xunit\xunit\src\xunit.core\Sdk\ExceptionAggregator.cs:line 96
[0.049s] [TestLifetime] [Information] Finished test DATA_Received_TooSlowlyOnSecondStream_AbortsConnectionAfterNonAdditiveRateTimeout in 0.0474257s
```
</details>

## Build

https://dev.azure.com/dnceng/public/_build/results?buildId=1269732&view=results

### LINKED ISSUE #34947: CanReadAndWriteWithAsyncConnectionMiddleware
## Failing Test(s)

- Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.ConnectionMiddlewareTests.CanReadAndWriteWithAsyncConnectionMiddleware

## Error Message

```text
Assert.Equal() Failure
↓ (pos 75)
Expected: ···750 09:51:25 GMT

Hello World!
Actual:   ···750 09:51:25 GMT

lo World!���
↑ (pos 75)
```

## Stacktrace

<details>

```text
   at Microsoft.AspNetCore.Testing.StreamBackedTestConnection.Receive(String[] lines) in /_/src/Servers/Kestrel/shared/test/StreamBackedTestConnection.cs:line 125
   at Microsoft.AspNetCore.Testing.StreamBackedTestConnection.ReceiveEnd(String[] lines) in /_/src/Servers/Kestrel/shared/test/StreamBackedTestConnection.cs:line 130
   at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.ConnectionMiddlewareTests.CanReadAndWriteWithAsyncConnectionMiddleware(RequestDelegate requestDelegate) in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/ConnectionMiddlewareTests.cs:line 83
   at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.ConnectionMiddlewareTests.CanReadAndWriteWithAsyncConnectionMiddleware(RequestDelegate requestDelegate) in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/ConnectionMiddlewareTests.cs:line 89
--- End of stack trace from previous location ---
```
</details>

## Logs

<details>

```text
[0.001s] [TestLifetime] [Information] Starting test CanReadAndWriteWithAsyncConnectionMiddleware-Microsoft.AspNetCore.Http.RequestDelegate at 2021-08-02T14:50:03
[0.001s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting starting
[0.002s] [Microsoft.Hosting.Lifetime] [Information] Now listening on: http://127.0.0.1:0
[0.002s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Debug] Loaded hosting startup assembly InMemory.FunctionalTests
[0.002s] [Microsoft.Hosting.Lifetime] [Information] Application started. Press Ctrl+C to shut down.
[0.002s] [Microsoft.Hosting.Lifetime] [Information] Hosting environment: Production
[0.002s] [Microsoft.Hosting.Lifetime] [Information] Content root path: /datadisks/disk1/work/9D11085F/w/9C6708E5/e/
[0.002s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting started
[0.002s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMALOJ655K30" accepted.
[0.002s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMALOJ655K30" started.
[0.002s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request starting HTTP/1.0 POST http:/// - 12
[0.002s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HMALOJ655K30", Request id "0HMALOJ655K30:00000002": started reading request body.
[0.003s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HMALOJ655K30", Request id "0HMALOJ655K30:00000002": done reading request body.
[0.003s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request finished HTTP/1.0 POST http:/// - 12 - 200 - - 0.2072ms
[0.003s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMALOJ655K30" disconnecting.
[0.003s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMALOJ655K30" stopped.
[0.077s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting stopping
[0.077s] [Microsoft.Hosting.Lifetime] [Information] Application is shutting down...
[0.077s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting stopped
[0.304s] [Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.ConnectionMiddlewareTests] [Error] Test threw an exception.
Xunit.Sdk.EqualException: Assert.Equal() Failure
                                     ↓ (pos 75)
Expected: ···750 09:51:25 GMT\r\n\r\nHello World!
Actual:   ···750 09:51:25 GMT\r\n\r\nlo World!\0\0\0
                                     ↑ (pos 75)
   at Xunit.Assert.Equal(String expected, String actual, Boolean ignoreCase, Boolean ignoreLineEndingDifferences, Boolean ignoreWhiteSpaceDifferences) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\StringAsserts.cs:line 244
   at Microsoft.AspNetCore.Testing.StreamBackedTestConnection.Receive(String[] lines) in /_/src/Servers/Kestrel/shared/test/StreamBackedTestConnection.cs:line 125
   at Microsoft.AspNetCore.Testing.StreamBackedTestConnection.ReceiveEnd(String[] lines) in /_/src/Servers/Kestrel/shared/test/StreamBackedTestConnection.cs:line 130
   at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.ConnectionMiddlewareTests.CanReadAndWriteWithAsyncConnectionMiddleware(RequestDelegate requestDelegate) in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/ConnectionMiddlewareTests.cs:line 83
   at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.ConnectionMiddlewareTests.CanReadAndWriteWithAsyncConnectionMiddleware(RequestDelegate requestDelegate) in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/ConnectionMiddlewareTests.cs:line 89
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 264
--- End of stack trace from previous location ---
   at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 48
   at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\Dev\xunit\xunit\src\xunit.core\Sdk\ExceptionAggregator.cs:line 90
[0.305s] [TestLifetime] [Information] Finished test CanReadAndWriteWithAsyncConnectionMiddleware-Microsoft.AspNetCore.Http.RequestDelegate in 0.3041502s
```
</details>

## Build

https://dev.azure.com/dnceng/public/_build/results?buildId=1271314

### LINKED ISSUE #34982: Failing Test `HttpLogging.FileLoggerProcessorTests.WritesToNewFileOnOptionsChange`
## Failing Test(s)

- Microsoft.AspNetCore.HttpLogging.FileLoggerProcessorTests.WritesToNewFileOnOptionsChange

## Error Message

```text
Assert.Equal() Failure
↓ (pos 0)
Expected: Message two

Actual:

↑ (pos 0)
```

## Stacktrace

<details>

```text
   at Microsoft.AspNetCore.HttpLogging.FileLoggerProcessorTests.WritesToNewFileOnOptionsChange() in /_/src/Middleware/HttpLogging/test/FileLoggerProcessorTests.cs:line 386
--- End of stack trace from previous location ---
```
</details>

## Logs

<details>

```text
[xUnit.net 00:00:02.26]     Microsoft.AspNetCore.HttpLogging.FileLoggerProcessorTests.WritesToNewFileOnOptionsChange [FAIL]
[xUnit.net 00:00:02.26]       Assert.Equal() Failure
[xUnit.net 00:00:02.26]                 â†“ (pos 0)
[xUnit.net 00:00:02.26]       Expected: Message two\n
[xUnit.net 00:00:02.26]       Actual:   
[xUnit.net 00:00:02.26]                 â†‘ (pos 0)
[xUnit.net 00:00:02.26]       Stack Trace:
[xUnit.net 00:00:02.26]         /_/src/Middleware/HttpLogging/test/FileLoggerProcessorTests.cs(386,0): at Microsoft.AspNetCore.HttpLogging.FileLoggerProcessorTests.WritesToNewFileOnOptionsChange()
[xUnit.net 00:00:02.26]         --- End of stack trace from previous location ---
[xUnit.net 00:00:03.08]   Finished:    Microsoft.AspNetCore.HttpLogging.Tests

```
</details>

## Build

https://dev.azure.com/dnceng/public/_build/results?buildId=1272695&view=ms.vss-test-web.build-test-results-tab&runId=37642180&resultId=108945&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab

### LINKED ISSUE #34986: Quarantine `FileLoggerProcessorTests.WritesToNewFileOnNewInstance`
## Failing Test(s)

- Microsoft.AspNetCore.HttpLogging.FileLoggerProcessorTests.WritesToNewFileOnNewInstance

## Error Message

```text
Assert.Equal() Failure
↓ (pos 0)
Expected: Message three

Actual:

↑ (pos 0)
```

## Stacktrace

<details>

```text
   at Microsoft.AspNetCore.HttpLogging.FileLoggerProcessorTests.WritesToNewFileOnNewInstance() in /_/src/Middleware/HttpLogging/test/FileLoggerProcessorTests.cs:line 330
--- End of stack trace from previous location ---

```
</details>

## Logs

<details>

```text
[xUnit.net 00:00:04.31]     Microsoft.AspNetCore.HttpLogging.FileLoggerProcessorTests.WritesToNewFileOnNewInstance [FAIL]
[xUnit.net 00:00:04.31]       Assert.Equal() Failure
[xUnit.net 00:00:04.31]                 â†“ (pos 0)
[xUnit.net 00:00:04.31]       Expected: Message three\n
[xUnit.net 00:00:04.31]       Actual:   
[xUnit.net 00:00:04.31]                 â†‘ (pos 0)
[xUnit.net 00:00:04.31]       Stack Trace:
[xUnit.net 00:00:04.32]         /_/src/Middleware/HttpLogging/test/FileLoggerProcessorTests.cs(330,0): at Microsoft.AspNetCore.HttpLogging.FileLoggerProcessorTests.WritesToNewFileOnNewInstance()
[xUnit.net 00:00:04.32]         --- End of stack trace from previous location ---
[xUnit.net 00:00:05.31]   Finished:    Microsoft.AspNetCore.HttpLogging.Tests

```
</details>

## Build

https://dev.azure.com/dnceng/public/_build/results?buildId=1271097&view=ms.vss-test-web.build-test-results-tab&runId=37609180&resultId=110685&paneView=debug

### LINKED ISSUE #34992: Quarantine Failing Linux Tests
## Failing Test(s)

- Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.RequestTests.Http10KeepAliveNotHonoredIfResponseContentLengthNotSet

## Error Message

```text
Assert.Equal() Failure
↓ (pos 75)
Expected: ···807 11:57:47 GMT

Goodbye
Actual:   ···807 11:57:47 GMT

e������
↑ (pos 75)
```

## Stacktrace

<details>

```text
   at Microsoft.AspNetCore.Testing.StreamBackedTestConnection.Receive(String[] lines) in /_/src/Servers/Kestrel/shared/test/StreamBackedTestConnection.cs:line 125
   at Microsoft.AspNetCore.Testing.StreamBackedTestConnection.ReceiveEnd(String[] lines) in /_/src/Servers/Kestrel/shared/test/StreamBackedTestConnection.cs:line 130
   at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.RequestTests.Http10KeepAliveNotHonoredIfResponseContentLengthNotSet() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs:line 751
   at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.RequestTests.Http10KeepAliveNotHonoredIfResponseContentLengthNotSet() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs:line 757
--- End of stack trace from previous location ---
```
</details>

## Logs

<details>

```text
[0.000s] [TestLifetime] [Information] Starting test Http10KeepAliveNotHonoredIfResponseContentLengthNotSet at 2021-08-03T12:51:57
[0.002s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting starting
[0.002s] [Microsoft.Hosting.Lifetime] [Information] Now listening on: http://127.0.0.1:0
[0.002s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Debug] Loaded hosting startup assembly InMemory.FunctionalTests
[0.002s] [Microsoft.Hosting.Lifetime] [Information] Application started. Press Ctrl+C to shut down.
[0.002s] [Microsoft.Hosting.Lifetime] [Information] Hosting environment: Production
[0.002s] [Microsoft.Hosting.Lifetime] [Information] Content root path: /root/helix/work/workitem/
[0.002s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting started
[0.003s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMAMFLPJRFGN" accepted.
[0.003s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMAMFLPJRFGN" started.
[0.003s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request starting HTTP/1.0 GET http:/// - -
[0.003s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMAMFLPJRFGN" completed keep alive response.
[0.003s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request finished HTTP/1.0 GET http:/// - - - 200 0 - 0.1069ms
[0.035s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request starting HTTP/1.0 POST http:/// - 7
[0.035s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HMAMFLPJRFGN", Request id "0HMAMFLPJRFGN:00000003": started reading request body.
[0.037s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HMAMFLPJRFGN", Request id "0HMAMFLPJRFGN:00000003": done reading request body.
[0.038s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request finished HTTP/1.0 POST http:/// - 7 - 200 - - 3.0973ms
[0.038s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMAMFLPJRFGN" disconnecting.
[0.038s] [Microsoft.AspNetCore.Server.Kestrel.Connections] [Debug] Connection id "0HMAMFLPJRFGN" stopped.
[0.038s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting stopping
[0.038s] [Microsoft.Hosting.Lifetime] [Information] Application is shutting down...
[0.039s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting stopped
[0.116s] [Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.RequestTests] [Error] Test threw an exception.
Xunit.Sdk.EqualException: Assert.Equal() Failure
                                     ↓ (pos 75)
Expected: ···807 11:57:47 GMT\r\n\r\nGoodbye
Actual:   ···807 11:57:47 GMT\r\n\r\ne\0\0\0\0\0\0
                                     ↑ (pos 75)
   at Xunit.Assert.Equal(String expected, String actual, Boolean ignoreCase, Boolean ignoreLineEndingDifferences, Boolean ignoreWhiteSpaceDifferences) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\StringAsserts.cs:line 244
   at Microsoft.AspNetCore.Testing.StreamBackedTestConnection.Receive(String[] lines) in /_/src/Servers/Kestrel/shared/test/StreamBackedTestConnection.cs:line 125
   at Microsoft.AspNetCore.Testing.StreamBackedTestConnection.ReceiveEnd(String[] lines) in /_/src/Servers/Kestrel/shared/test/StreamBackedTestConnection.cs:line 130
   at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.RequestTests.Http10KeepAliveNotHonoredIfResponseContentLengthNotSet() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs:line 751
   at Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.RequestTests.Http10KeepAliveNotHonoredIfResponseContentLengthNotSet() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs:line 757
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 273
--- End of stack trace from previous location ---
   at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 54
   at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\Dev\xunit\xunit\src\xunit.core\Sdk\ExceptionAggregator.cs:line 96
[0.117s] [TestLifetime] [Information] Finished test Http10KeepAliveNotHonoredIfResponseContentLengthNotSet in 0.1162396s
```
</details>

## Build

https://dev.azure.com/dnceng/public/_build/results?buildId=1272938&view=ms.vss-test-web.build-test-results-tab&runId=37646266&resultId=107222&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes https://github.com/dotnet/aspnetcore/issues/34903
Fixes https://github.com/dotnet/aspnetcore/issues/34685
Fixes https://github.com/dotnet/aspnetcore/issues/34947
Fixes https://github.com/dotnet/aspnetcore/issues/34982
Fixes https://github.com/dotnet/aspnetcore/issues/34986
Fixes https://github.com/dotnet/aspnetcore/issues/34992

### CODE REVIEW

**Review by BrennanConroy — APPROVED** [review]
Nice!

File: src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3ConnectionTests.cs [hunk 1]
```diff
@@ -262,7 +262,6 @@ public async Task SETTINGS_MaxFieldSectionSizeSent_ServerReceivesValue()
         }
 
         [Fact]
-        [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/34685")]
```
Comment (Tratcher) [inline src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3ConnectionTests.cs:265]: There are several more tests in this file linking to the same issue. Are they all ready to unquarantined?

Comment (sebastienros) [reply src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3ConnectionTests.cs:265]: Will check

Comment (sebastienros) [reply src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3ConnectionTests.cs:265]: You were right. They were not mentioned in the issue and I failed to notice them. Will check the other files too then.

**Pull request merged**